File: Emlak/src/app/admin/model/IDosya.ts
export interface IDosya {
    ID: number,
    //[Required(ErrorMessage = "Baþlýk alaný boþ olamaz ve en fazla 50 karakter olmalýdýr.")]
    //[StringLength(50)]
    Title: string,
    //[DataType(DataType.MultilineText)]
    //[AllowHtml]
    Description: string,
    FileUrl: string,
    Code: string,
    Active: boolean,
    Deleted: boolean,

    Mesaj: string,

    OldFileUrl: string,
    HasFile: boolean,

    TypesList: string[]
}

File: AdminPanel/Repository/Dosya/IDosya.cs
using System.Collections.Generic;
using System.Web.Mvc;

namespace Repository.DosyaModel
{
    public interface IDosya
    {
        int ID { get; set; }
        string Title { get; set; }
        string Description { get; set; }
        string FileUrl { get; set; }
        string Code { get; set; }
        bool Active { get; set; }
        bool Deleted { get; set; }

        string Mesaj { get; set; }

        List<SelectListItem> TypesList { get; set; }

        List<Dosya> List();
        IDosya Select(int id);
        bool Insert(IDosya dosya);
        bool Update(IDosya dosya);
        bool Delete(int id);
        bool Remove(int id);
    }
}
